src/python/pants/engine/internals/graph.py

```diff
@@ -568,18 +568,81 @@ def __init__(
         )
 
 
+class InvalidFileDependency(Exception):
+    pass
+
+
+class ParsedDependencies(NamedTuple):
+    addresses: List[Address]
+    files: List[str]
+
+
+def parse_dependencies_field(
```

Comment:
FYI this code all gets refactored in the upcoming ignore PR. Same heuristics will apply, though.